functions/Copy-SqlIntegrationServices.ps1

```diff
@@ -0,0 +1,392 @@
+            $f = $destinationFolders | ? { $_.Name -eq $Folder }
+            if ($force) {
+                $f.Environments[$Environment].Drop()
+                $f.Alter()
+                $f.Refresh()
+            }
+            $srcEnv = ($sourceFolders | ? { $_.Name -eq $Folder }).Environments[$Environment]
+            $targetEnv = New-Object "$ISNamespace.EnvironmentInfo" ($f, $($srcEnv.Name), $($srcEnv.Description))
+            foreach ($var in $srcEnv.Variables) {
+                if ($var.Value.ToString() -eq "") { 
+                    $finalValue= ""
+                }
+                else {
+                    $finalValue = $var.Value
+                }
+                $targetEnv.Variables.Add($var.Name, $var.Type, $finalValue, $var.Sensitive, $var.Description)
+            }
+            $targetEnv.Create()
+            $targetEnv.Alter()
+            $targetEnv.Refresh()
+        }
+    }
+    PROCESS
+    {
+        if (!$sourceCatalog) {
+            Write-Error "The source SSISDB catalog does not exist."
+            exit
+        }
+        if (!$destinationCatalog) {
+            # Todo, prompt to create the catalog?
+            Write-Error "The destination SSISDB catalog does not exist."
+            exit
+        }
+        if ($folder) {
+            if ($($sourceFolders.Name) -contains $folder) {
+                $srcFolder = $sourceFolders | ? { $_.Name -eq $folder }
+                if ($($destinationFolders.Name) -contains $folder) {
+                    if (!$force) {
+                        Write-Warning "Integration services catalog folder $folder exists at destination. Use -Force to drop and recreate."
+                    }
+                    else {
+                        If ($Pscmdlet.ShouldProcess($Destination, "Dropping folder $folder and recreating")) {
+                            try {
+                                Create-Folder -Folder $($srcFolder.Name) -Description $($srcFolder.Description) -Force
+                            }
+                            catch {
+                                Write-Exception $_
+                            }
+                        
+                        }
+                    }
+                }
+                else {
+                    If ($Pscmdlet.ShouldProcess($Destination, "Creating folder $folder")) {
+                        try {
+                            Create-Folder -Folder $($srcFolder.Name) -Description $($srcFolder.Description)
+                        }
+                        catch {
+                            Write-Exception $_
+                        }
+                    }
+                }
+            }
+            else {
+                Write-Error "The source folder provided does not exist in the source Integration Services catalog."
+            }
+        }
+        else {
+            foreach ($srcFolder in $sourceFolders) {
+                if($($destinationFolders.Name) -notcontains $($srcFolder.Name)) {  
```

Comment:
You don't have to $() here - plain ol' $destinationFolders.Name would work. I outlined below where the $() is needed.